public/js/store.js

```diff
@@ -0,0 +1,150 @@
+/*-------------------------------------------------------------------------------------------------------------------*\
+ |  Copyright (C) 2015 PayPal                                                                                          |
+ |                                                                                                                     |
+ |  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance     |
+ |  with the License.                                                                                                  |
+ |                                                                                                                     |
+ |  You may obtain a copy of the License at                                                                            |
+ |                                                                                                                     |
+ |       http://www.apache.org/licenses/LICENSE-2.0                                                                    |
+ |                                                                                                                     |
+ |  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed   |
+ |  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for  |
+ |  the specific language governing permissions and limitations under the License.                                     |
+ \*-------------------------------------------------------------------------------------------------------------------*/
+
+var Dispatcher = require('./dispatcher');
+var Constants = require('./constants');
+var EventEmitter = require('events').EventEmitter;
+var assign = require('object-assign');
+var _model = {products: [], cart: {}};
+var $ = require('jquery');
+
+function createProduct(productObject, callback) {
+	$.ajax({
+		url: '/products',
+		type: 'POST',
+		dataType: 'json',
+		data: productObject,
+		cache: false,
+		success: function (data) {
+			console.log('data', data);
+			_model.products = data.products;
+			callback();
+		},
+		error: function (xhr, status, err) {
+			console.error(err.toString());
+		}
+	});
+}
+function deleteProduct(productObject, callback) {
+	$.ajax({
+		url: '/products',
+		type: 'POST',
```

Comment:
Also while answering this comment I found that I introduced a bug (which has now been fixed) where I wasn't passing the `_method` parameter.. So thanks for that as well :)